### ISSUE TITLE
feature request: enforce comments after last import

### ISSUE THREAD

**Issue by TrigamDev** [import-js/eslint-plugin-import#3226]
At least personally, I often use a comment separator between imports and the rest of the file, especially when the file has a large number of imports. Having a rule to enforce this would be very helpful.

This could potentially conflict with `newline-after-import`, though. I can think of two ways to resolve this.

# Solution 1
The separator comment must be placed either before or after the new lines enforced by  `newline-after-import` . Whether the comment is placed before or after could potentially be configured with an option in the rule, or one could be chosen to be hardcoded with it not being configurable. This is a little clunky, however.

### Options
```json5
{
	"import/comment-after-import": [
		"warn",

		/* Whether to style as line comments,
		or a block comment. */
		// "line" | "block"
		"block",

		/* The content of the comment.
		Can be one string using \n, or an array of
		strings, one element representing a line. */
		// string | string[]
		"---------------",

		/* Whether the comment should be placed
		before or after the new lines enforced by
		newline-after-import. */
		// "before" | "after"
		"before"
	],

	"import/newline-after-import": [
		"warn",
		{
			count: 2
		}
	]
}
```

### ❌  Invalid
```ts
import foo from "bar"
import type { baz } from "quux"


console.log("Hello world!")
```

### ✅  Valid
```ts
import foo from "bar"
import type { baz } from "quux"
/* --------------- */


console.log("Hello world!")
```

# Solution 2
Instead of adding a new `comment-after-import` rule, `newline-after-import` could be modified to enforce lines with specific content, not just empty lines. The rule would also have to be renamed to reflect this change.

### Options
```json5
{
	"import/lines-after-import": [
		"warn",
		{
			/* The lines that should be present after
			the last import.
			Can be one string using \n, or an array of
			strings, one element representing a line. */
			// string | string[]
			"lines": [
				"",
				"",
				"/* --------------- */",
				"",
				""
			]
		}
	]
}
```

### ❌  Invalid
```ts
import foo from "bar"
import type { baz } from "quux"


console.log("Hello world!")
```

### ✅  Valid
```ts
import foo from "bar"
import type { baz } from "quux"


/* --------------- */


console.log("Hello world!")
```

**Comment by ljharb** [comment]
I'm a bit confused why, in a world with syntax highlighting, a comment is helpful to differentiate between imports and regular code?

**Comment by TrigamDev** [comment]
> I'm a bit confused why, in a world with syntax highlighting, a comment is helpful to differentiate between imports and regular code?

Sorry, didn't see the reply until now.
At least personally, I get a little overwhelmed when it comes to larger files, and having comment dividers to split up different sections helps a lot with this. It's not about telling what is and isn't imports, but visually breaking up the two into distinct sections.

**Comment by ludgerey** [comment]
I've never seen that in any real code. Why bother having this in a linter?

Personally I'd oppose having this in my code as well, because this comment has no value at all a side from visual effect.

I'd suggest picking an IDE that helps visualizing stuff like that better if you struggle with it @TrigamDev

**Comment by KAMRONBEK** [comment]
@TrigamDev in case it helps while this proposal is discussed: with ESLint 9 flat config you can get this behavior today as a ~35-line **local rule**, no plugin change needed. Tested (report + autofix) on eslint 9:

```js
// eslint.config.mjs
const commentAfterImport = {
  meta: {
    type: 'layout',
    fixable: 'code',
    schema: [{
      type: 'object',
      properties: {
        comment: { type: 'string' },        // separator text
        style: { enum: ['line', 'block'] }, // // ... or /* ... */
      },
      additionalProperties: false,
    }],
    messages: {
      missing: 'Expected a "{{comment}}" separator comment after the last import.',
    },
  },
  create(context) {
    const { comment = '---------------', style = 'block' } = context.options[0] ?? {};
    const sourceCode = context.sourceCode;
    return {
      'Program:exit'(program) {
        const imports = program.body.filter((n) => n.type === 'ImportDeclaration');
        if (imports.length === 0) return;
        const last = imports[imports.length - 1];
        const found = sourceCode.getCommentsAfter(last).some((c) => c.value.trim() === comment.trim());
        if (found) return;
        const text = style === 'line' ? `// ${comment}` : `/* ${comment} */`;
        context.report({
          node: last,
          messageId: 'missing',
          data: { comment },
          fix: (fixer) => fixer.insertTextAfter(last, `\n${text}`),
        });
      },
    };
  },
};

export default [
  {
    plugins: { local: { rules: { 'comment-after-import': commentAfterImport } } },
    rules: {
      'local/comment-after-import': ['warn', { comment: '---------------', style: 'block' }],
    },
  },
];
```

It composes with `import/newline-after-import` (the separator sits directly after the last import; the newline rule's `considerComments` option controls how blank lines interact with trailing comments). Covers your Solution 1 shape — line/block style and the separator text are the options; extending it to multi-line separators or a before/after-blank-line placement option is straightforward from here.
